logger.ts

```diff
@@ -0,0 +1,36 @@
+import fetch from 'cross-fetch';
+const _debounce = require('lodash/debounce');
+
+const ingestEndpoint = process.env.AXIOM_INGEST_ENDPOINT || process.env.NEXT_PUBLIC_AXIOM_INGEST_ENDPOINT;
+
+const debouncedSendLogs = _debounce(() => sendLogs(), 1000);
+
+let collectedLogs: any[] = [];
+
+function _log(level: string, message: string, args: any = {}) {
+  const l = { level, message, ...args };
+  // TODO: print to console?
+  console.log(l);
+  collectedLogs.push(l);
+  debouncedSendLogs();
+}
+
+export const log = {
+  debug: (message: string, args: any = {}) => _log('debug', message, args),
+  info: (message: string, args: any = {}) => _log('info', message, args),
+  warn: (message: string, args: any = {}) => _log('warn', message, args),
+  error: (message: string, args: any = {}) => _log('error', message, args),
+};
+
+function sendLogs() {
+  const url = `${ingestEndpoint}/?type=log`;
```

Comment:
This is actually more complicated as the ingest endpoint currently has the `?type=web-vitals` in it. We can't really remove that as that might break existing WebVital installations, so I think the way to go here is parsing the `ingestEndpoint` as an URL and setting the type parameter.

Reply:
done, what do you think?

Reply:
Love it